### PR TITLE
Basic Token Details Page

### PR DESCRIPTION
**Pull request by youngkidwarrior**
WIP token details page

Ethen wanted to show this to users. Should be clean enough for now.

Running into a lot of problems because `TopNav` doesn't ergonomically support switching the nav with query params 

Future PRs will address the shortcomings with TopNav and further flesh out the feature set of this page

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/309?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#309** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/309?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#305** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/305?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#299** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/299?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by 0xBigBoss** [comment]
It's pretty easy to get 429'd by coingecko. we should add that stale time increase or figure out how to have react query cache it.

Also, not sure if it was due to the errors, but it started showing the object Object
Screenshot 2024-04-20 at 17 56 01

**Review by 0xBigBoss — COMMENTED** [review]
It's pretty easy to get 429'd by coingecko. we should add that stale time increase or figure out how to have react query cache it.

Also, not sure if it was due to the errors, but it started showing the object Object
Screenshot 2024-04-20 at 17 56 01

File: packages/app/components/TopNav.tsx [hunk 1]
```diff
@@ -73,8 +75,15 @@ export function TopNav({
...
   }
 
+  const [tokenParam, setTokenParam] = useToken()
+  const hasSubrouteParam = tokenParam !== undefined
+
   const handleBack = () => {
     // always pop to the base path. e.g. /account/settings/edit-profile -> /account
+    if (hasSubrouteParam) {
+      setTokenParam(undefined)
+      return
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:85]: the return is back lol

File: packages/app/components/TopNav.tsx [hunk 2]
```diff
@@ -84,17 +93,45 @@ export function TopNav({
 
   const isSubRoute = !noSubroute && parts.length > 1
 
+  const selectedCoin = coins.find((c) => c.token === tokenParam)
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:96]: wait ami hallucinating, wasn't this in another PR already?

File: packages/app/components/TopNav.tsx (hunk 1 above)
Comment (youngkidwarrior) [reply packages/app/components/TopNav.tsx:85]: Fixed in a later PR

File: packages/app/components/TopNav.tsx (hunk 2 above)
Comment (youngkidwarrior) [reply packages/app/components/TopNav.tsx:96]: Fixed in a later PR

**Review by 0xBigBoss — APPROVED** [review]

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Apr 22, 5:21 PM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/309).

**Pull request merged**